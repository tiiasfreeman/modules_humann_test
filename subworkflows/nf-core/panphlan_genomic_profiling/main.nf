//
// Run PanPhlAn: Metagenomic profiling to identify gene composition of individual strains
//

include {   PANPHLAN_DOWNLOADPANGENOME              }   from '/workspace/modules_humann_test/modules/nf-core/panphlan/downloadpangenome/main.nf'
include {   PANPHLAN_MAP                            }   from '/workspace/modules_humann_test/modules/nf-core/panphlan/panphlan_map/main.nf'
include {   PANPHLAN_PROFILING                      }   from '/workspace/modules_humann_test/modules/nf-core/panphlan/panphlan_profiling/main.nf'

workflow PANPHLAN_GENOMIC_PROFILING {

take:
    val_species_name                    //   value: "Genus_species";                    name of species for analysis
    ch_sample_sequences                  // channel: val(meta), path(list of files);     meta: ID is final name for run file, fastq sequence(s) for samples
    //val_single_end                      //   value: "[True/False]";                     [True] single-end samples, [False] paired-end samples

    main:

    ch_versions = Channel.empty()

    PANPHLAN_DOWNLOADPANGENOME (val_species_name)
    ch_versions = ch_versions.mix(PANPHLAN_DOWNLOADPANGENOME.out.versions.first())

    def sample_sequences = ch_sample_sequences
        .flatMap{ metadata, files -> files.collect {file -> [metadata, file] } }
        //.flatten()

    sample_sequences.each { tuple_sequence ->
        PANPHLAN_MAP (
            PANPHLAN_DOWNLOADPANGENOME.out.pangenome,
            PANPHLAN_DOWNLOADPANGENOME.out.indexes,
            tuple_sequence,
            val_species_name)
        }
    ch_versions = ch_versions.mix(PANPHLAN_MAP.out.versions.first())

    panphlan_merged_maps_txt = PANPHLAN_MAP.out.mapping_dir
        .map{ it[1].toString() }
        .collectFile(name: 'panphlan_merged_maps.txt') { paths -> paths.join('\n')}
    panphlan_merged_maps_txt.view()

    PANPHLAN_PROFILING (
        panphlan_merged_maps_txt,
        PANPHLAN_DOWNLOADPANGENOME.out.pangenome,
        val_species_name)
    ch_versions = ch_versions.mix(PANPHLAN_MAP.out.versions.first())

emit:

    pangenome               = PANPHLAN_DOWNLOADPANGENOME.out.pangenome          // channel: $prefix/{species_name}_pangenome.tsv; pangenome used for selected species
    mapping_directory       = PANPHLAN_MAP.out.mapping_dir                      // channel: val(meta), path("${species_name}_map"); mapping directory built for selected species
    panphlan_merged_maps_txt
    panphlan_profile        = PANPHLAN_PROFILING.out.profile_matrix             // channel: val(meta), path("*.tsv"); final profile matrix

    versions                = ch_versions                                       // channel: [ versions.yml ]
}
