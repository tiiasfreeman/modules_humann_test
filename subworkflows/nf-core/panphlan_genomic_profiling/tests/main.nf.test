nextflow_workflow {

    name "Subworkflow PANPHLAN_GENOMIC_PROFILING"
    script "subworkflows/nf-core/panphlan_genomic_profiling/main.nf"
    config "./nextflow.config"
    workflow "PANPHLAN_GENOMIC_PROFILING"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/panphlan_genomic_profiling"
    tag "panphlan/downloadpangenome"
    tag "panphlan/panphlan_map"
    tag "panphlan/panphlan_profiling"

    test("Bacteroides_fragilis - combined single end - fails") {
        when {
            workflow {
                """
                input[0] = "Bacteroides_fragilis"
                input[1] = Channel.of([ [ id:'test', single_end:false ], // meta map
                        [file("https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz", checkIfExists:true),
                        file("https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz", checkIfExists:true),
                        file("https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test2_1.fastq.gz", checkIfExists:true),
                        file("https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test2_2.fastq.gz", checkIfExists:true)]
                    ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
