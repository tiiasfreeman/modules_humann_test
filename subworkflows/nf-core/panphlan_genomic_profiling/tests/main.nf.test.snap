{
    "Bacteroides_fragilis - combined single end - fails": {
        "content": [
            {
                "0": [
                    "Bacteroides_fragilis_pangenome.tsv:md5,1efc51187d49ef33f23894f64c6c2d47"
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    "versions.yml:md5,91040bc18a58ef8d1f8deb19414b74cf"
                ],
                "mapping_directory": [
                    
                ],
                "pangenome": [
                    "Bacteroides_fragilis_pangenome.tsv:md5,1efc51187d49ef33f23894f64c6c2d47"
                ],
                "panphlan_merged_maps_txt": [
                    
                ],
                "panphlan_profile": [
                    
                ],
                "versions": [
                    "versions.yml:md5,91040bc18a58ef8d1f8deb19414b74cf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-13T08:18:15.821873583"
    }
}